pennylane/qnode.py

```diff
@@ -963,9 +963,14 @@ def __call__(self, *args, **kwargs) -> qml.typing.Result:
             if "mode" in self.execute_kwargs:
                 self.execute_kwargs.pop("mode")
             # pylint: disable=unexpected-keyword-arg
+            if self.transform_program.is_empty():
+                transform_program = None
+            else:
+                transform_program = self.transform_program

```

Comment:
If we make empty transform programs valid transform programs, we will no longer need this line.

Sensible defaults can help eliminate the need for custom edge case logic.